metrics/ci_integrations/lib/client/github_actions/github_actions_client.dart

```diff
@@ -0,0 +1,256 @@
+        return bodyParser(body);
+      } else {
+        final reason = response.body == null || response.body.isEmpty
+            ? response.reasonPhrase
+            : response.body;
+        return InteractionResult.error(
+          message: 'Failed to perform an operation with code '
+              '${response.statusCode}. Reason: $reason',
+        );
+      }
+    } catch (error) {
+      return InteractionResult.error(
+        message: 'Failed to perform an operation. Error details: $error',
+      );
+    }
+  }
+
+  /// Retrieves a list of workflow runs by the given [workflowFileName].
+  ///
+  /// Results with a [List] of [WorkflowRun]s.
+  ///
+  /// [status] can be used to set the status of workflow runs to retrieve.
+  /// [status] defaults to [RunStatus.completed].
+  ///
+  /// [perPage] can be used to set the number of runs per page.
+  /// [perPage] defaults to `100` runs per page.
+  /// The maximum value of [perPage] is 100 runs per page.
+  ///
+  /// [page] can be used to set the number of a page to retrieve.
+  /// [page] defaults to the `1`st page.
+  Future<InteractionResult<List<WorkflowRun>>> fetchWorkflowRuns(
+    String workflowFileName, {
+    RunStatus status = RunStatus.completed,
+    int perPage = 100,
+    int page = 1,
+  }) {
+    const statusMapper = RunStatusMapper();
+
+    final queryParameters = {
+      'status': statusMapper.unmap(status),
+      'per_page': '$perPage',
+      'page': '$page',
+    };
+
+    final url = UrlUtils.buildUrl(
+      baseUrl,
+      path: 'workflows/$workflowFileName/runs',
+      queryParameters: queryParameters,
+    );
+
+    return _handleResponse<List<WorkflowRun>>(
+      _client.get(url, headers: headers),
+      (Map<String, dynamic> json) {
+        final runList =
+            json == null ? null : json['workflow_runs'] as List<dynamic>;
+
+        return InteractionResult.success(
+          result: WorkflowRun.listFromJson(runList),
+        );
+      },
+    );
+  }
+
+  /// Retrieves a workflow run duration by the given [runId].
+  ///
+  /// Results with a [WorkflowRunDuration].
+  Future<InteractionResult<WorkflowRunDuration>> fetchRunDuration(int runId) {
+    final url = UrlUtils.buildUrl(
+      baseUrl,
+      path: 'runs/$runId/timing',
+    );
+
+    return _handleResponse<WorkflowRunDuration>(
+      _client.get(url, headers: headers),
+      (Map<String, dynamic> json) {
+        return InteractionResult.success(
+          result: WorkflowRunDuration.fromJson(json),
+        );
+      },
+    );
+  }
+
+  /// Retrieves a list of workflow run artifacts by the given [runId].
+  ///
+  /// Results with a [List] of [WorkflowRunArtifact]s.
+  ///
+  /// [perPage] can be used to set the number of artifacts per page.
+  /// [perPage] defaults to `100` artifacts per page.
+  /// The maximum value of [perPage] is 100.
+  ///
+  /// [page] can be used to set the number of a page to retrieve.
+  /// [page] defaults to the `1`st page.
```

Comment:
```suggestion
  /// Fetches a list of artifacts for a workflow run with the given [runId].
  ///
  /// [perPage] can be used to set the number of artifacts per page.
  /// [perPage] defaults to `100` artifacts per page.
  /// The maximum value of [perPage] is 100.
  ///
  /// [page] can be used to set the number of a page to retrieve.
  /// [page] defaults to the `1`st page.
```
For parameters, the same comment as in a similar method.